TechnicalDocuments/CompomentComposition.md

```diff
@@ -0,0 +1,63 @@
+# Bento Component Composition Using `UIStackView`
+
+#### Updated at: 23-3-2019
+
+## The scope of this document 
+Component Composition with `Bento` is a very wide topic due to its flexible design.
+`Bento` components can be composed in many ways.
+
+The purpose of this document is to provide documentation for __composing Bento components using `UIStackView`__.
+Other ways of composition are out of the scope of this document.
+
+## How to use component composition in your framework
+
+Well, the answer to this question is that your can't.
+The reasoning for this decision is that
+* we must reuse components via the `DesignLibrary`
+* If you need something custom and customizable then using `UIStackView`s probably won't be good enough for you
+* We need to be flexible and to depend on abstractions.
+
+#### Where and how to use component composition
+In `BabylonUI` there is a new namespace called `UI` which is the place for `Atomic Components`.
+
+- `Atomic Components` are the `Bento` Components version of `UIKit`'s components so the nomenclature is similar  like `UILabel` and `UI.Label`, `UIButton` and `UI.Button` etc 
+- `Atomic Components` use `Stylesheets` for styling
+- `Atomic Components` should behave like their `UIKit` counterpart.
+
+## Code Sample
+
+```swift
+[
+    UI.Label(
+      text: "Some Title",
+      styleSheet: LabelStyleSheet()
+          .compose(\.textColor, tokens.colors.black)
+          .compose(\.font, font(fontAttributes: tokens.typography.body))
+    ).asAnyRenderable(),
+    UI.Label(
+        text: "Some Details",
+        styleSheet: LabelStyleSheet()
+            .compose(\.textColor, tokens.colors.grey700)
+            .compose(\.font, font(fontAttributes: tokens.typography.subhead))
+    ).asAnyRenderable()
+]
+.stack(
+    axis: .vertical,
+    spacing: 2
+)
```

Comment:
Would be nice also to have an image of how it's going to look like. We could make a table e.g

| Code | Screenshot |
| --- | --- |
| some code | screenshot |

Reply:
How can we have something like this?

`enter` creates a new line we would have to create new lines with `<br>` which isn't neither convenient nor scalable.

I believe that the `GalleryApp` that we have is good, its real working code after all :)

Reply:
> I believe that the GalleryApp that we have is a good solution for now, its real working code after all :)
 
Not really, if it forces people to dig through code every time they want to do something. In a way it defeats the purpose of documentation if you defer with "check the code."

Reply:
The `GalleryApp` is written in such a way which is similar with Xcode playgrounds.

Every component is in a different group/folder like 

|--- Rows
|--- ItemPicker
|--- Toggle

and within these folders there is only *one* file with content similar to this

```swift
struct ItemPickerBuilder {
    func make() -> UIViewController {
        return ComponentPreviewerBuilder().make { library in
            [
                library.componentsBuilder()
                    .itemPicker(
                        title: "Country",
                        subtitle: "UK"
                    )
            ]
        }
    }
}
```

So yes you "check the code" but its not really that you have to dig in order to find it.

Reply:
> Would be nice also to have an image of how it's going to look like. We could make a table e.g

Here I mean that when we provide a composition example in the documentation to have a screenshot an the side to show how it's going to look like